src/aspire/utils/filters.py

```diff
@@ -73,6 +75,21 @@ def evaluate_grid(self, L, *args, **kwargs):
 
         return h
 
+    def dual(self):
+        return DualFilter(copy.deepcopy(self))
```

Comment:
Why is a copy needed? Is there a risk of the filter changing?

Reply:
The `deepcopy` returns an independent copy of original one. So the created DualFilter object is independent. Originally we worry about that it get changed If the original one is modified. If we like them paired together, them we should remove the `deepcopy`.

Reply:
Ok, but again, is there a risk of the filter changing? In other words, can the original filter object change? Is it mutable? I was under the impression that it was not.

Reply:
Actually CFT filter can be changed such as by downsample, so we should not use deepcopy if we want to make dual filter keep updated with the original one.

Reply:
@janden `deepcopy` was removed and the branch was rebased.

Reply:
Ok. In that case we need to fix the `Filter` classes to be immutable. It will be very confusing for the dual filter to update if we downsample the original one. Will make an issue.